hibernate-cache/spring-local/src/main/java/org/infinispan/tutorial/simple/hibernate/cache/spring/local/InfinispanHibernateCacheSpringLocal.java

```diff
@@ -0,0 +1,266 @@
+package org.infinispan.tutorial.simple.hibernate.cache.spring.local;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.stat.SecondLevelCacheStatistics;
+import org.hibernate.stat.Statistics;
+import org.infinispan.tutorial.simple.hibernate.cache.spring.local.model.Event;
+import org.infinispan.tutorial.simple.hibernate.cache.spring.local.model.Person;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.TypedQuery;
+import java.util.List;
+
+@SpringBootApplication
+public class InfinispanHibernateCacheSpringLocal {
+
+   private static final Logger log = LoggerFactory.getLogger(InfinispanHibernateCacheSpringLocal.class);
+
+   @Autowired
+   private EntityManagerFactory emf;
+
+   @Bean
+   public CommandLineRunner demo() {
```

Comment:
For me personally, this tutorial is just too complicated. I would prefer to put it into the demos and replace the code here with something really simple, like putting an entity using `EntityManager` and obtaining it from Infinispan (or just check if it's there).

Reply:
Hibernate does not expose internal Infinispan details to users. There's no way to achieve that and we don't even want to encourage that.

I think the tutorial is thorough but not complicated. It deals with very simple/common things for Hibernate users: entities, queries and caching concepts (such as expiring data from the cache)